### ISSUE TITLE
Duplicate check is kinda pointless right now

### ISSUE THREAD

**Issue by Shade-** [MyBBStuff/MyAlerts#265]
The current duplicate check is smart, but doesn't prevent the actual user case it was designed for in the first place. For what I see, MyAlerts can typically send multiple notifications for events such as replies in a thread. In a board with a mention plugin such as DVZ Mentions installed, a simple reply to a subscribed thread with a mention delivers 3 separate notifications when it really should deliver only one.

https://github.com/MyBBStuff/MyAlerts/blob/24f00f7bf2edcce1bcf047ae88d497482ca9cb8f/inc/plugins/MybbStuff/MyAlerts/src/AlertManager.php#L301-303

This can be improved by stripping out the alert code from the array key.

```
			static::$alertQueue[$alert->getUserId() . '_' . $alert->getObjectId(
			)] = $alert;
```

**Comment by euantorano** [comment]
Yep, this is a known problem. There are two approaches I’ve considered to this:

1. Alert “stacking” - multiple alerts for one piece of content, grouped together and displayed as one - eg, you were quoted and mentioned in your subscribed thread X
2. Prevent duplicates, with the highest priority alert type being the one that gets through

I’m considering simplifying the blocking code slightly to make it less clever so that it inserts straight away, with a query before to check for a duplicate.
